lesson4/argirovatina_hometask.rb

```diff
@@ -0,0 +1,46 @@
+class LazyHash < Hash
+
+  def method_missing(method_name)
+    method_name = get_correct_key(method_name)
+    if self.keys.include? method_name
+      self[method_name] = self[method_name].call
+    else
+      raise "Method #{method_name} is missing"
+    end
+  end
+
+  def lazy?(key)
+    key = get_correct_key(key)
+    self[key].respond_to? :call
+  end
+
+  def respond_to_missing?(method_name, include_private = false)
+    method_name = get_correct_key(method_name)
```

Comment:
you should not raise error if method_name wasn't found as a key in our LazyHash. in that case we just need to return `false` from `#respond_to_missing?`
```ruby
  get_correct_key(method_name) rescue
```